### PR TITLE
[Cache] removeAll: remove all files in cache path

### LINKED ISSUE #125: removeAll doesn't remove UIKit extension formats before using them
Original post from @PFloding:

> BTW, I had to try to delete the disk cache to force URL requests, and in the process I may have found an issue.
> I resorted to calling Haneke.sharedImageCache.removeAll()
> But this only worked if I had first accessed the cache at least once. And it attempted to delete a non-existent directory before doing the correct one.
> Is there a correct way to delete the persistent cache?

### PR DESCRIPTION
**Pull request by hpique**
Remove all files in cache path no matter if the format was registered or not. Fixes #125.

### CODE REVIEW

File: Haneke/Cache.swift [hunk 1]
```diff
@@ -129,13 +129,32 @@ public class Cache<T: DataConvertible where T.Result == T, T : DataRepresentable
...
             memoryCache.removeAllObjects()
-            diskCache.removeAllData()
+            dispatch_group_enter(group)
+            diskCache.removeAllData {
+                dispatch_group_leave(group)
+            }
+        }
+        dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0)) {
+            let timeout = dispatch_time(DISPATCH_TIME_NOW, Int64(60 * NSEC_PER_SEC))
+            dispatch_group_wait(group, timeout);
```
Comment (dcharbonnier) [inline Haneke/Cache.swift]: you don't want to check the return value ? at least for log ?

**Comment by dcharbonnier** [comment]
this is suspicious :

```
override func tearDown() {
        sut.removeAll()
        super.tearDown()
    }
```
I think it should be 

```
override func tearDown() {
  sut.removeAll() {
    super.tearDown()
  }
}

File: Haneke/Cache.swift (hunk 1 above)
Comment (hpique) [inline Haneke/Cache.swift]: @dcharbonnier Good call.

**Comment by hpique** [comment]
`super.tearDown` should always be called within `tearDown`. This should do the trick:

```swift
override func tearDown() {
    let semaphore = dispatch_semaphore_create(0)
    sut.removeAllData() {
        dispatch_semaphore_signal(semaphore)
    }
    dispatch_semaphore_wait(semaphore, DISPATCH_TIME_FOREVER)
    super.tearDown()
}
```

**Comment by dcharbonnier** [comment]
right, removeAllData or removeAll ?

**Comment by hpique** [comment]
Both. Same problem in DiskCacheTests.

**Comment by hpique** [comment]
Actually, what I posted is wrong as it would block the main queue. Threading is hard. Will revisit in a bit.

**Pull request merged**